sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -283,6 +286,8 @@ class CSVOptions(
     charToEscapeQuoteEscaping.foreach(format.setCharToEscapeQuoteEscaping)
     if (isCommentSet) {
       format.setComment(comment)
+    } else if (legacyDefaultUnicodeNullAsWrittenComment) {
```

Comment:
If [univocity-parsers#518](https://github.com/uniVocity/univocity-parsers/pull/518) is merged, the code can be like:
`else {
    writerSettings.setQuoteCommentStartingFirstColumnEnabled(false)
}`